server/src/main/java/org/elasticsearch/action/admin/indices/resolve/ResolveClusterAction.java

```diff
@@ -0,0 +1,455 @@
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.util.concurrent.CountDown;
+import org.elasticsearch.common.util.concurrent.EsExecutors;
+import org.elasticsearch.core.Strings;
+import org.elasticsearch.index.IndexNotFoundException;
+import org.elasticsearch.search.SearchService;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.ConnectTransportException;
+import org.elasticsearch.transport.NoSeedNodeLeftException;
+import org.elasticsearch.transport.NoSuchRemoteClusterException;
+import org.elasticsearch.transport.RemoteClusterAware;
+import org.elasticsearch.transport.RemoteClusterService;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.elasticsearch.action.search.TransportSearchHelper.checkCCSVersionCompatibility;
+
+/**
+ * Action to provide information about clusters to help with cross-cluster search scenarios.
+ *
+ * The action accepts an index expression that can include remote clusters. Exclusion notation
+ * for clusters is also permitted. For example: logs*,*:logs*,-remote7:*, will query
+ * the local cluster ("logs*"), all configured remote clusters except for the cluster with
+ * alias "remote7".
+ *
+ * The response will indicate for each cluster whether it is connected to the querying cluster,
+ * what the skip_unavailable setting is for that cluster, version info about each cluster
+ * and whether it has any indices, aliases or datastreams that match the index expression passed in.
+ *
+ * To determine whether any indices, aliases or datastreams match the index expression passed in
+ * on each cluster, the ResolveIndexAction functionality is used. ResolveIndexAction provides a
+ * complete listing of all matching indices, aliases or datastreams, whereas this endpoint only
+ * provides a boolean value about whether anything matches at all. For detailed information
+ * about which indices, aliases and/or datastreams match the _resolve/index endpoint should be used.
+ */
+public class ResolveClusterAction extends ActionType<ResolveClusterAction.Response> {
```

Comment:
There's no need to subclass `ActionType` here. It's a common pattern (again, it made more sense in the distant past) but we're trying to remove it (see e.g.  #103055, #102800, #102354, #101053, #101037).

Reply:
Fixed in [this commit](https://github.com/elastic/elasticsearch/pull/102726/commits/d1f6d81226d3416685abaae358a936369963675d).

Since, following the ResolveIndexAction structure, that is the top level class, I decided to deconstruct all of these inner classes to separate ones and follow the pattern of the field-caps endpoint.

Reply:
I think there's still cleanup to do here, this class should not be `extends ActionType<ResolveClusterAction.Response>` and it looks like `ResolveClusterAction.TransportAction` and `TransportResolveClusterAction` are mostly duplicates.

Reply:
Yeah, sorry, this class was meant to be deleted. Will be deleted in the next commit.